Wip windows

Osyotr opened the pull request.

Osyotr committed:
Fix some invalid usages of isspace and isdigit

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

Osyotr committed:
Always use vendored fast_double_parser for now

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

Osyotr committed:
WIP Timey stuff

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

Osyotr committed:
Add CMakePresets.json and integrate with vcpkg

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

Osyotr committed:
WIP windows action

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

vaanwd commented:
Hello!
Do you have some instrunctions of how to build Organic Maps on Windows?
Thank`s!

Osyotr commented:
The build steps are there: https://github.com/Osyotr/organicmaps/blob/wip-windows/.github/workflows/windows-check.yaml
The important step is to skip some submodules.

PS: windows builds are not usable at this moment due to #7838

vaanwd commented:
> The build steps are there: https://github.com/Osyotr/organicmaps/blob/wip-windows/.github/workflows/windows-check.yaml The important step is to skip some submodules.
> 
> PS: windows builds are not usable at this moment due to [organicmaps#7838](https://github.com/organicmaps/organicmaps/issues/7838)

Yes, i have the same problem(Black screen) in Arch Linux with QT 6.8.
There is a posibility to downgrade QT in vcpkg?

Osyotr commented:
Yes but I think the OpenGL issue is deeper because with Qt 6.6 it will crash 9/10 times.
https://learn.microsoft.com/en-us/vcpkg/consume/lock-package-versions?tabs=inspect-powershell#5---force-a-specific-version

Osyotr committed:
tmp

Osyotr committed:
Always use forward slashes on Windows

Windows is performing path normalization anyways.
Fixes some unit tests that expect the output to contain forward slashes.

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

Osyotr committed:
Fix some unit tests on Windows

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

Osyotr committed:
WIP protobuf

Osyotr committed:
Fix QOpenGLContext creation on Windows

Osyotr committed:
Don't set geometry for MainWindow

It's going to be maximized anyway.

hb0nd commented:
Hiya @Osyotr,
Thank you very much for working on this! It'd be great to merge it to CoMaps - what do you think? 
https://codeberg.org/comaps/comaps/issues/30

Osyotr commented:
This PR is not ready to be merged yet. At the very least, dependency management and CI pipeline should be done in a way that won't disturb existing linux/osx/android/ios workflows. I can't really tell when I'll be able to finish this, but I certainly won't stop anyone from using these changes as a base for their own work.

BTW can CoMaps cherry-pick commits from OrganicMaps and vice versa?

hb0nd commented:
> This PR is not ready to be merged yet. At the very least, dependency management and CI pipeline should be done in a way that won't disturb existing linux/osx/android/ios workflows. I can't really tell when I'll be able to finish this, but I certainly won't stop anyone from using these changes as a base for their own work.

No worries at all, thank you for working on it :)

> BTW can CoMaps cherry-pick commits from OrganicMaps and vice versa?

Yup! We're keeping Apache v2 licensing for now so it's possible (AFAIK it requires attribution, we have it in CoMaps already).

vng committed:
[feature] Removed IndexFactory class.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

^ Conflicts:
^	libs/indexer/data_source.cpp
^	libs/indexer/feature.cpp
^	libs/indexer/features_vector.cpp
^	libs/indexer/mwm_set.hpp

vng committed:
[feature] Put m_metaDeserializer into SharedLoadInfo.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[feature] Added Feature V1 version with free Header2 bits.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[feature] Added RouteRelation and linked it with FeatureType.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[coding] Rename non-empty string serialization.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[base] Updated IsSortedAndUnique function.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[base] Added to_uint base for string_view.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[coding] Added varint short arrays ser/des.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

# Conflicts:
#	libs/coding/coding_tests/varint_test.cpp

vng committed:
Include fix.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

AndrewShkrob committed:
[style] Some improvements

Signed-off-by: Andrei Shkrob <github@shkrob.dev>

vng committed:
[drape] Added hatching selector.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[styles] Wetland hatching colors and priority.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
Draw Bookmark's names.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

Osyotr committed:
test action

vng committed:
[tracks] Fixed starting points.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

biodranik commented on transit/transit_schedule.cpp:
```diff
@@ -15,7 +15,11 @@ uint32_t constexpr kMask6bits = 0x3F;
 {
   std::tm tm;
+#ifdef _MSC_VER
+  localtime_s(&tm, &ts);
```
> As with all bounds-checked functions, localtime_s is only guaranteed to be available if __STDC_LIB_EXT1__ is defined by the implementation and if the user defines __STDC_WANT_LIB_EXT1__ to the integer constant 1 before including [<time.h>](https://en.cppreference.com/w/c/header/time.html).

Add these defines explicitly and use the same function on all platforms?

Osyotr replied:
Они тупо не реализованы в glibc. https://godbolt.org/z/51E5hfsqx

biodranik commented on 3party/CMakeLists.txt:
```diff
@@ -63,11 +63,17 @@ add_subdirectory(opening_hours)
 add_subdirectory(stb_image)
 add_subdirectory(succinct)
 
+add_library(fast_double_parser INTERFACE)
```
А это для чего? Может вообще заменить парсер более быстрой версией https://github.com/fastfloat/fast_float

Osyotr replied:
Это я писал для того, чтобы убрать из `base/CMakeLists.txt` захардкоженный путь к `3party`, используя таргет `fast_double_parser::fast_double_parser`, чтобы оно в конечном итоге нормально дружило с `WITH_SYSTEM_PROVIDED_3PARTY`.
Но засовывать это под `WITH_SYSTEM_PROVIDED_3PARTY` было бессмысленно - таргет `fast_double_parser::fast_double_parser` появился только в https://github.com/lemire/fast_double_parser/commit/ca05d13e26cc8e42acf6d3196a9d943f7aea9531 и эти правки не успели разползтись по поддерживаемым дистрибутивам.

biodranik replied:
Мне не нравится добавление _медленно обновляемых маинтайнерами_ зависимостей в дистрибутивах. Мы готовы достаточно быстро и оперативно обновлять всё, что нужно, и тащить как можно меньше системных зависимостей. Это сильно упрощает всё.

biodranik commented on 3party/opening_hours/opening_hours_supported_features_tests/opening_hours_supported_features_tests.cpp:
```diff
@@ -6,11 +6,29 @@
 int countTests = 0;
 
+#ifdef _MSC_VER
+extern "C" char * strptime(char const * s, char const * f, std::tm * tm)
```
Можно ли переписать использования этой функции на https://en.cppreference.com/w/cpp/chrono/parse.html ?

biodranik commented on 3party/vulkan_wrapper/CMakeLists.txt:
```diff
@@ -22,4 +22,9 @@ if (PLATFORM_LINUX)
 # dlopen
-target_link_libraries(${PROJECT_NAME} $<$<BOOL:CMAKE_DL_LIBS>:${CMAKE_DL_LIBS}>)
+if(WIN32)
+  find_package(dlfcn-win32 CONFIG REQUIRED)
```
Зачем dlopen на винде? В коде вулкана везде используется ::LoadLibraryA:
```
#  if defined( _WIN32 )
          m_library = ::LoadLibraryA( vulkanLibraryName.c_str() );
#  elif defined( __unix__ ) || defined( __APPLE__ ) || defined( __QNX__ ) || defined( __Fuchsia__ )
          m_library = dlopen( vulkanLibraryName.c_str(), RTLD_NOW | RTLD_LOCAL );

```

biodranik commented on CMakeLists.txt:
```diff
@@ -191,6 +192,9 @@ endif()
 
 find_package(ZLIB REQUIRED)
 
+find_package(Boost CONFIG REQUIRED)
```
У меня буст подтягивался из 3party, там сейчас не работает?

Osyotr replied:
В этом PR вместо `3party` большинство библиотек берутся из vcpkg. Эти правки переносить пока рано.

biodranik commented on base/atomic_shared_ptr.hpp:
```diff
@@ -16,11 +16,11 @@ class AtomicSharedPtr final
 
-  void Set(ValueType value) noexcept { atomic_store(&m_wrapped, value); }
-  ValueType Get() const noexcept { return atomic_load(&m_wrapped); }
+  void Set(ValueType value) noexcept { m_wrapped.store(value); }
```
Аккуратно допилим https://github.com/organicmaps/organicmaps/pull/9910 ?

Osyotr replied:
ИМХО, лучше обходить atomic_shared_ptr стороной.

biodranik commented on base/base_tests/beam_tests.cpp:
```diff
@@ -32,7 +32,7 @@ void Smoke()
   vector<double> expected;
   for (size_t i = 0; i < kCapacity; ++i)
-    expected.emplace_back(kTotal - 1 - i);
+    expected.emplace_back(static_cast<double>(kTotal - 1 - i));
```
Так сработает?
```suggestion
    expected.emplace_back(kTotal - 1.0 - i);
```

Osyotr replied:
Не уверен, тут по сути то же самое, что было раньше: неявное приведение `size_t` к `double`.

biodranik commented on base/file_name_utils.cpp:
```diff
@@ -58,11 +58,7 @@ string GetDirectory(string const & name)
-#else
-    return '/';
-#endif
+  return '/';
```
?

Osyotr replied:
В сообщении к коммиту описал, почему так.
https://github.com/Osyotr/organicmaps/commit/a7bec677f0d58d6b14aea08390fb4e3151b6f8b3

biodranik replied:
Верно, нашёл упоминания о нормализации в fopen и std::filesystem. Спасибо, не знал.

biodranik commented on base/random.hpp:
```diff
@@ -11,7 +11,9 @@ template <class T> class UniformRandom
   std::mt19937 m_gen;
-  std::uniform_int_distribution<T> m_distr;
+
+  using distribution_int_type = std::conditional_t<sizeof(T) != 1, T, std::conditional_t<std::is_signed_v<T>, short, unsigned short>>;
```
Что тут был за косяк?

Osyotr replied:
https://github.com/organicmaps/organicmaps/pull/8502#pullrequestreview-2216257804

vng committed:
[tracks] Fixed ending points.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[tracks] Set close distance threshold to 10m.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

kavikhalique committed:
Implement Track Selection and elevation info display on PP

Signed-off-by: kavikhalique <kavikhalique3@gmail.com>

vng committed:
[legal] Updated NOTICE and README.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

biodranik committed:
[linux] Fix zero file creation (birth) time

Observed in Ubuntu 24 via Orb on arm-based Mac

Signed-off-by: Alexander Borsuk <me@alex.bio>

Osyotr committed:
Add windows support to vulkan_wrapper

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

Osyotr committed:
Make UniformRandom standard-conforming

std::uniform_int_distribution<T> requires sizeof(T) >= sizeof(short)

Signed-off-by: Osyotr <Osyotr@users.noreply.github.com>

AndrewShkrob committed:
[ci] Add PR write perms to code style workflow

Signed-off-by: Andrei Shkrob <github@shkrob.dev>

Ferenc- committed:
[libs][linux] Fix incorrect include of fcntl.h

The POSIX standard says, that `fcntl.h` is to be included as `#include <fcntl.h>`.

This change fixes the compiler warnings that happen with `musl` libc :
```
/usr/include/sys/fcntl.h:1:2: warning: #warning redirecting incorrect #include <sys/fcntl.h> to <fcntl.h> [-Wcpp]
    1 | #warning redirecting incorrect #include <sys/fcntl.h> to <fcntl.h>
      |  ^~~~~~~
In file included from /home/ferenc/github.com/organicmaps/organicmaps/libs/coding/mmap_reader.cpp:18,
                 from /home/ferenc/github.com/organicmaps/organicmaps/build-alpine-3.21/build-alpine-3.21/libs/coding/CMakeFiles/coding.dir/Unity/unity_0_cxx.cxx:40:
/usr/include/sys/fcntl.h:1:2: warning: #warning redirecting incorrect #include <sys/fcntl.h> to <fcntl.h> [-Wcpp]
    1 | #warning redirecting incorrect #include <sys/fcntl.h> to <fcntl.h>
      |  ^~~~~~~
````

This happens because of the explicit [warning in the proxy header](
https://git.musl-libc.org/cgit/musl/tree/include/sys/fcntl.h)

Signed-off-by: Ferenc Géczi <ferenc.gm@gmail.com>

weblate committed:
[strings] Update translations

Co-authored-by: Adolfo Jayme Barrientos <fitojb@ubuntu.com>
Co-authored-by: Alexander Gabilondo <alexgabi@openmailbox.org>
Co-authored-by: Andrei “Talkerbox” Audzei <bigtalkerbox@gmail.com>
Co-authored-by: Dan <jonweblin2205@protonmail.com>
Co-authored-by: Drugi Sapog <dindrugi@users.noreply.hosted.weblate.org>
Co-authored-by: ERYpTION <eryption@users.noreply.hosted.weblate.org>
Co-authored-by: Emin Tufan Çetin <etcetin@gmail.com>
Co-authored-by: Fabrizio Zaini <f.zaini@alice.it>
Co-authored-by: Federico Zotti <34000603+Fede-26@users.noreply.github.com>
Co-authored-by: Fjuro <fjuro@alius.cz>
Co-authored-by: Halbast Abdullah <halbastabdullah7@gmail.com>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Jaŭhien <j.kisliakou@gmail.com>
Co-authored-by: Jim Kats <jim-kats@hotmail.com>
Co-authored-by: Luca <luca@decode.lc>
Co-authored-by: Marko Damnjanovic <markodamnjanovic93@gmail.com>
Co-authored-by: Mickaël Binos <mickaelbinos@outlook.com>
Co-authored-by: Priit Jõerüüt <jrthwlate@users.noreply.hosted.weblate.org>
Co-authored-by: Sveinn í Felli <sv1@fellsnet.is>
Co-authored-by: The Cats <philosoph@danwin1210.de>
Co-authored-by: Thiago Hilario <thiagolm.hilario@gmail.com>
Co-authored-by: Weblate Translation Memory <noreply-mt-weblate-translation-memory@weblate.org>
Co-authored-by: Yurt Page <yurtpage@gmail.com>
Co-authored-by: boa-conflictor <dragoonares@protonmail.com>
Co-authored-by: fin-w <fin-w@users.noreply.hosted.weblate.org>
Co-authored-by: hulse <ealasa@tutamail.com>
Co-authored-by: lp <lp+we@ik.me>
Co-authored-by: summoner001 <summoner@vivaldi.net>
Signed-off-by: Adolfo Jayme Barrientos <fitojb@ubuntu.com>
Signed-off-by: Alexander Gabilondo <alexgabi@openmailbox.org>
Signed-off-by: Andrei “Talkerbox” Audzei <bigtalkerbox@gmail.com>
Signed-off-by: Dan <jonweblin2205@protonmail.com>
Signed-off-by: Drugi Sapog <dindrugi@users.noreply.hosted.weblate.org>
Signed-off-by: ERYpTION <eryption@users.noreply.hosted.weblate.org>
Signed-off-by: Emin Tufan Çetin <etcetin@gmail.com>
Signed-off-by: Fabrizio Zaini <f.zaini@alice.it>
Signed-off-by: Federico Zotti <34000603+Fede-26@users.noreply.github.com>
Signed-off-by: Fjuro <fjuro@alius.cz>
Signed-off-by: Halbast Abdullah <halbastabdullah7@gmail.com>
Signed-off-by: Jaŭhien <j.kisliakou@gmail.com>
Signed-off-by: Jim Kats <jim-kats@hotmail.com>
Signed-off-by: Luca <luca@decode.lc>
Signed-off-by: Marko Damnjanovic <markodamnjanovic93@gmail.com>
Signed-off-by: Mickaël Binos <mickaelbinos@outlook.com>
Signed-off-by: Priit Jõerüüt <jrthwlate@users.noreply.hosted.weblate.org>
Signed-off-by: Roman Tsisyk <roman@tsisyk.com>
Signed-off-by: Sveinn í Felli <sv1@fellsnet.is>
Signed-off-by: The Cats <philosoph@danwin1210.de>
Signed-off-by: Thiago Hilario <thiagolm.hilario@gmail.com>
Signed-off-by: Weblate Translation Memory <noreply-mt-weblate-translation-memory@weblate.org>
Signed-off-by: Yurt Page <yurtpage@gmail.com>
Signed-off-by: boa-conflictor <dragoonares@protonmail.com>
Signed-off-by: fin-w <fin-w@users.noreply.hosted.weblate.org>
Signed-off-by: hulse <ealasa@tutamail.com>
Signed-off-by: lp <lp+we@ik.me>
Signed-off-by: summoner001 <summoner@vivaldi.net>
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/af/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/ar/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/ast/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/az/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/ca/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/cs/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/da/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/el/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/en/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/es/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/es_MX/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/et/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/fa/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/fi/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/fr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/gl/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/he/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/hi/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/hu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/is/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/it/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/ja/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/ko/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/mr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/nl/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/pt/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/ru/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/sk/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/sr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/th/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/tr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/uk/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/vi/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/zh_Hant_HK/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/ast/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/ca/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/ckb/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/cs/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/cy/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/el/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/es/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/es_MX/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/et/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/eu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/fr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/gl/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/hu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/is/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/it/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/nl/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/tr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/uk/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/ast/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/ca/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/es/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/et/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/tr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plist/es_MX/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/ru/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/ast/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/ca/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/cs/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/da/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/el/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/en/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/es/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/es_MX/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/et/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/fa/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/fr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/gl/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/hu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/is/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/it/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/nl/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/pt/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/ru/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/sr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/tr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/uk/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/ast/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/ca/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/es/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/es_MX/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/eu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/hu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/it/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/tts/tr/

kirylkaveryn committed:
[ios] Disable user interaction on PP vc close

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

kirylkaveryn committed:
[ios] Disable track recording live activity widget for apps running on macos

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

kirylkaveryn committed:
[ios] Check that pp data exists before update the TR PP screen

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

kirylkaveryn committed:
[ios] Log error during the track editing instead of failing

The bug is quite rare and happens when the user taps on the edit (pencil) button on the PlacePage. It seems like the tap happens during the PP data object reloading the and the object type has changed. It is better to avoid failing in such cases because we cannot prevent user interaction during the pp reloading.

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

vng committed:
Minor fixes.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
Added Spline::Equidistant.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[drape] Added basic routes drawing.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
Separate license for the binary map data files (.mwm)

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
[tracks] Safe GpsTrackCollection requests during recording.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

dvdmrtnz committed:
[strings] Add more Spanish categories translations

Signed-off-by: David Martinez <47610359+dvdmrtnz@users.noreply.github.com>

biodranik committed:
[ios] Fixed ATM translation in the PlacePage

Signed-off-by: Alexander Borsuk <me@alex.bio>

biodranik committed:
[ios] Fix opening websites with percent encoding characters

Signed-off-by: Alexander Borsuk <me@alex.bio>

biodranik committed:
[TTS] Clearer pronouncement of route recalculation in ru

> При уходе с тропы если язык подсказок русский, говорит - «преращик маршрута».

Signed-off-by: Alexander Borsuk <me@alex.bio>

weblate committed:
[strings] Update translations

Co-authored-by: Andi Chandler <andi@gowling.com>
Co-authored-by: Andrei “Talkerbox” Audzei <bigtalkerbox@gmail.com>
Co-authored-by: Daniele Ziglioli <d.ziglioli@gmail.com>
Co-authored-by: David Martinez <47610359+dvdmrtnz@users.noreply.github.com>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Mohammad Abu Garbeyyeh <optics-pitch.1z@icloud.com>
Co-authored-by: Quentin <mrbydix@users.noreply.hosted.weblate.org>
Co-authored-by: Weblate Translation Memory <noreply-mt-weblate-translation-memory@weblate.org>
Co-authored-by: jonnysemon <jonnysemon@users.noreply.hosted.weblate.org>
Signed-off-by: Andi Chandler <andi@gowling.com>
Signed-off-by: Andrei “Talkerbox” Audzei <bigtalkerbox@gmail.com>
Signed-off-by: Daniele Ziglioli <d.ziglioli@gmail.com>
Signed-off-by: David Martinez <47610359+dvdmrtnz@users.noreply.github.com>
Signed-off-by: Mohammad Abu Garbeyyeh <optics-pitch.1z@icloud.com>
Signed-off-by: Quentin <mrbydix@users.noreply.hosted.weblate.org>
Signed-off-by: Weblate Translation Memory <noreply-mt-weblate-translation-memory@weblate.org>
Signed-off-by: jonnysemon <jonnysemon@users.noreply.hosted.weblate.org>
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/es/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/it/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/ar/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/fr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/he/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/ar/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/en_GB/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/es/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/it/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/he/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/tts/en_GB/

vng committed:
[styles] Regenerated.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

biodranik committed:
Some translations for Hiking and Cycling layers

Signed-off-by: Alexander Borsuk <me@alex.bio>

biodranik committed:
[android] Hiking and cycling layers

Signed-off-by: Alexander Borsuk <me@alex.bio>

biodranik committed:
[ios] Hiking and cycling layers

Signed-off-by: Alexander Borsuk <me@alex.bio>

biodranik committed:
[qt][core] Hiking and cycling layers

Signed-off-by: Alexander Borsuk <me@alex.bio>

kavikhalique committed:
Removed animated slow rendering and Nit picks

Signed-off-by: kavikhalique <kavikhalique3@gmail.com>

hemanggs committed:
[android] Refactoring regression fix

Signed-off-by: hemanggs <hemangmanhas@gmail.com>

biodranik committed:
iOS ui for bookmarks text placement

Signed-off-by: Alexander Borsuk <me@alex.bio>

biodranik committed:
Bookmark placement translations

Signed-off-by: Alexander Borsuk <me@alex.bio>

vng committed:
[tools] Script to propagate En l11s for all iOS Localizable files.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

AndrewShkrob committed:
[android-auto] Fix null pointer exception

Signed-off-by: Andrei Shkrob <github@shkrob.dev>

vng committed:
[strings][iOS] Propagated En for missing translations.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

kirylkaveryn committed:
[ios] Fix track recording indicator image

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

kirylkaveryn committed:
[ios] Fix map overlay buttons UI; add new icons

1. Squash the MWMMapOverlayManagerObserver methods into the one that handles all the states. The each method for each button is overhead and is redundant.
2. Icons for the hiking and cycling
3. Slightly update style of the layers button

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

kirylkaveryn committed:
[bookmark] Switch bookmark's title outline on map Light/Dark style change

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

biodranik committed:
[android] Show bookmarks names setting

Signed-off-by: Alexander Borsuk <me@alex.bio>

weblate committed:
[strings] Update translations

Co-authored-by: Alexander Gabilondo <alexgabi@openmailbox.org>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Jaŭhien <j.kisliakou@gmail.com>
Co-authored-by: Jim Kats <jim-kats@hotmail.com>
Co-authored-by: Sketch6580 <sketch6580@users.noreply.hosted.weblate.org>
Co-authored-by: Максим Горпиніч <gorpinicmaksim0@gmail.com>
Signed-off-by: Alexander Gabilondo <alexgabi@openmailbox.org>
Signed-off-by: Jaŭhien <j.kisliakou@gmail.com>
Signed-off-by: Jim Kats <jim-kats@hotmail.com>
Signed-off-by: Sketch6580 <sketch6580@users.noreply.hosted.weblate.org>
Signed-off-by: Максим Горпиніч <gorpinicmaksim0@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android-typestrings/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/el/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/eu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/uk/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/countries/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-typestrings/be/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/el/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/uk/

vng committed:
[android] Update active route icons.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

vng committed:
Show a toast when routes need maps update.

Signed-off-by: Viktor Govako <viktor.govako@gmail.com>

kirylkaveryn committed:
Add  initital `routes_update_maps_text toast` message string

Signed-off-by: Kiryl Kaveryn <kirylkaveryn@gmail.com>

weblate committed:
[strings] Update translations

Co-authored-by: Ghost of Sparta <makesocialfoss32@keemail.me>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Jim Kats <jim-kats@hotmail.com>
Co-authored-by: Sveinn í Felli <sv1@fellsnet.is>
Co-authored-by: Weblate Translation Memory <noreply-mt-weblate-translation-memory@weblate.org>
Co-authored-by: Максим Горпиніч <gorpinicmaksim0@gmail.com>
Signed-off-by: Ghost of Sparta <makesocialfoss32@keemail.me>
Signed-off-by: Jim Kats <jim-kats@hotmail.com>
Signed-off-by: Sveinn í Felli <sv1@fellsnet.is>
Signed-off-by: Weblate Translation Memory <noreply-mt-weblate-translation-memory@weblate.org>
Signed-off-by: Максим Горпиніч <gorpinicmaksim0@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/el/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/hu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/is/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/android/uk/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/af/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/ar/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/da/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/hi/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/ia/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/id/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/ja/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/ro/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/sv/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/tr/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios-plurals/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/el/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/hu/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/is/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/ios/uk/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/tts/en_GB/
Translate-URL: https://hosted.weblate.org/projects/organicmaps/tts/ia/
